File: DataStructures/graph/graphRepresentation.cpp
#include <bits/stdc++.h>
using namespace std;

void addEdge(std::vector<int> adj[],int u,int v){
	adj[u].push_back(v);
	adj[v].push_back(u);
}

void printGraph(vector<int> adj[],int V){
	for (int i = 0; i < V; ++i)
	{
		cout << "Adjacency list of vertex " << i << endl;
		cout << i << " ";
		for (auto x: adj[i])
		{
			cout << x << " ";
		}
		cout << endl;
	}
}

int main(int argc, char const *argv[])
{
	int V = 5;
	vector<int> adj[V];
	addEdge(adj,0,1);
	addEdge(adj,0,4);
	addEdge(adj,1,2);
	addEdge(adj,1,3);
	addEdge(adj,1,4);
	addEdge(adj,2,3);
	addEdge(adj,3,4);
	printGraph(adj,V);
	return 0;
}

File: DataStructures/graph/graphRepresentation.py
class AdjNode():
	"""docstring for AdjNode"""
	def __init__(self, data):
		super(AdjNode, self).__init__()
		self.vertex = data
		self.next = None
		


class Graph():
	"""docstring for Graph"""
	def __init__(self, vertices):
		super(Graph, self).__init__()
		self.V = vertices
		self.graph = [None]*self.V

	def add_edge(self,src,dest):
		node = AdjNode(dest)
		node.next = self.graph[src]
		self.graph[src] = node

		node = AdjNode(src)
		node.next = self.graph[dest]
		self.graph[dest] = node

	def print_graph(self):
		for i in range(self.V):
			print("Adjacency list of vertex {}\n head".format(i), end="")
			temp = self.graph[i]
			while temp:
				print(" -> {}".format(temp.vertex),end="")
				temp = temp.next
			print("\n")
		

if __name__ == '__main__':
	V = 5
	graph = Graph(V)
	graph.add_edge(0,1);
	graph.add_edge(0,4);
	graph.add_edge(1,2);
	graph.add_edge(1,3);
	graph.add_edge(1,4);
	graph.add_edge(2,3);
	graph.add_edge(3,4);

	graph.print_graph()